stl/inc/iterator

```diff
@@ -155,8 +176,8 @@ public:
     }
 
 protected:
-    _Container* container; // pointer to container
-    typename _Container::iterator iter; // iterator into container
+    _Container* container              = nullptr;
+    typename _Container::iterator iter = {};

```

Comment:
```suggestion
    typename _Container::iterator iter{};
```

Is this friendlier to the Cpp17DefaultConstructible requirements? The Standard depicts `ranges::iterator_t<Container> iter = ranges::iterator_t<Container>();` which is explicit default construction of a temporary followed by copy-init copy-construction. Saying `= {}` is implicit default construction. Saying `{}` is explicit default construction.

(I no longer pretend to understand C++ initialization. I guess I traded that for the relative simplicity of floating-point.)

Reply:
The form in the standard - `T t = T();` - is the most-generic form we have to express value-initialization. (There's no copy/move here since C++17: this simply initializes `t` with the prvalue `T()` per [dcl.init]/17.6.1.) We needed that fully generic form here before LWG-3149 fixed the `default_initializable` concept to require `T{}` to be valid.

I used copy-list init here because I had the impression you disliked mixing direct and copy init, not for any technical reason.

Reply:
I very mildly prefer consistency but copy-init of scalars doesn't really bother me.